pykusto/expressions.py

```diff
@@ -592,52 +602,116 @@ def assign(expression: ExpressionType, *columns: 'Column') -> 'AssignmentBase':
 
 
 class AssignmentToSingleColumn(AssignmentBase):
-    def __init__(self, column: 'Column', expression: ExpressionType) -> None:
+    def __init__(self, column: 'AnyTypeColumn', expression: ExpressionType) -> None:
         super().__init__(column.kql, expression)
 
 
 class AssignmentFromColumnToColumn(AssignmentToSingleColumn):
-    def __init__(self, target: 'Column', source: 'Column') -> None:
+    def __init__(self, target: 'AnyTypeColumn', source: 'BaseColumn') -> None:
         super().__init__(target, source)
 
 
 class AssignmentToMultipleColumns(AssignmentBase):
-    def __init__(self, columns: Union[List['Column'], Tuple['Column']], expression: ArrayType) -> None:
+    def __init__(self, columns: Union[List['AnyTypeColumn'], Tuple['AnyTypeColumn']], expression: ArrayType) -> None:
         super().__init__(KQL('({})'.format(', '.join(c.kql for c in columns))), expression)
 
 
 class AssignmentFromAggregationToColumn(AssignmentBase):
-    def __init__(self, column: Optional['Column'], aggregation: AggregationExpression) -> None:
+    def __init__(self, column: Optional['AnyTypeColumn'], aggregation: AggregationExpression) -> None:
         super().__init__(None if column is None else column.kql, aggregation)
 
 
-class Column(AnyExpression):
+class BaseColumn(BaseExpression):
     _name: str
 
+    def __new__(cls, *args, **kwargs):
+        if cls is 'BaseColumn':
+            raise TypeError("BaseColumn is abstract")
```

Comment:
use ABC

Reply:
AFAIK there is no way to achieve this behavior with ABC

Reply:
See this reply: https://github.com/Azure/pykusto/pull/54#discussion_r389370701